README.md

```diff
@@ -485,63 +547,63 @@ import ApiModel
+    dynamic var userId = ApiId()
+    dynamic var url = String() // generated on the server
 
-  var imageData: NSData?
+    var imageData: NSData?
 
-  class func apiConfig(config: ApiConfig) -> ApiConfig {
-    // ApiRequest.FormDataEncoding is where the magic happens
-    // It tells ApiModel to encode everything with `multipart/form-data`
-    config.encoding = ApiRequest.FormDataEncoding
-    return config
-  }
-
-  // Important because the `imageData` property cannot be represented by Realm
-  override class func ignoredProperties() -> [String] {
-    return ["imageData"]
-  }
-
-  override class func primaryKey() -> String {
-    return "userId"
-  }
-
-  class func apiNamespace() -> String {
-    return "user_avatar"
-  }
-
-  class func apiRoutes() -> ApiRoutes {
-    return ApiRoutes.resource("/user/avatar.json")
-  }
-
-  class func fromJSONMapping() -> JSONMapping {
-    return [
-      "userId": ApiIdTransform(),
-      "url": StringTransform()
-    ]
-  }
-
-  func JSONDictionary() -> [String:AnyObject] {
-    return [
-      "image": FileUpload(fileName: "avatar.jpg", mimeType: "image/jpg", data: imageData!)
-    ]
-  }
+    class func apiConfig(config: ApiConfig) -> ApiConfig {
+        // ApiRequest.FormDataEncoding is where the magic happens
+        // It tells ApiModel to encode everything with `multipart/form-data`
+        config.encoding = ApiRequest.FormDataEncoding
+        return config
+    }
+
+    // Important because the `imageData` property cannot be represented by Realm
+    override class func ignoredProperties() -> [String] {
+        return ["imageData"]
+    }
+
+    override class func primaryKey() -> String {
+        return "userId"
+    }
+
+    class func apiNamespace() -> String {
+        return "user_avatar"
+    }
+
+    class func apiRoutes() -> ApiRoutes {
+        return ApiRoutes.resource("/user/avatar.json")
+    }
+
+    class func fromJSONMapping() -> JSONMapping {
+        return [
+            "userId": ApiIdTransform(),
+            "url": StringTransform()
+        ]
+    }
+
+    func JSONDictionary() -> [String:AnyObject] {
+        return [
+            "image": FileUpload(fileName: "avatar.jpg", mimeType: "image/jpg", data: imageData!)
+        ]
+    }
 }
 
 func upload() {
-  let image = UIImage(named: "me.jpg")!
+    let image = UIImage(named: "me.jpg")!
 
-  let userAvatar = UserAvatar()
-  userAvatar.userId = "1"
-  userAvatar.imageData = UIImageJPEGRepresentation(image, 1)!
+    let userAvatar = UserAvatar()
+    userAvatar.userId = "1"
+    userAvatar.imageData = UIImageJPEGRepresentation(image, 1)!
 
-  Api(model: userAvatar).save { form in
-    if form.hasErrors {
-      print("Could not upload file: " + form.errorMessages.joinWithSeparator("\n"))
-    } else {
-      print("File uploaded! URL: \(userAvatar.url)")
+    Api(model: userAvatar).save { response in
+        if response.hasInternalServerError || response.hasValidationErrors || response.hasError {
```

Comment:
This is the stuff we want to avoid. Usability-wise it should really be "if success? else error?". In most cases you won't need to distinguish between server error or user error, as in my opinion, it would be to easy to ignore one of the two cases.